### PR TITLE
fix #44: refresh chain before printing current balance

### LINKED ISSUE #44: skepticoin.balance reports outdated information
skepticoin.balance doesn't refresh the chain.

So the balance reported will be incorrect after receiving coins, or after mining on a separate machine with the same wallet.

### PR DESCRIPTION
**Pull request by atramos**
For best user experience, it is recommended to also approve the other PRs.

### CODE REVIEW

File: skepticoin/scripts/utils.py [hunk 1]
```diff
@@ -107,6 +107,7 @@ def start_networking_peer_in_background(
     port: Optional[int] = None if args.dont_listen else args.listening_port
     thread = NetworkingThread(coinstate, port)
     thread.local_peer.network_manager.disconnected_peers = load_peers()
+    thread.daemon = True
```
Comment (sashimi-houdini) [inline skepticoin/scripts/utils.py:110]: I don't remember exactly why, but I chose a different idiom for closing the various threads. See e.g. `repl.py` for how it was done. Feel free to copy/paste that approach... though factoring it out is even better I suppose.

Now, if you can explain to me why the `thread.daemon` way is better, I might be persuaded to go that route, but in that case it should be in a separate PR (which includes said explanation) and should also be extended to all other locations where we currently have the try/finally approach from `repl.py`

**Comment by sashimi-houdini** [comment]
Other than the comment about `thread.daemon` and the flake errors this is good to go in.

File: skepticoin/scripts/utils.py (hunk 1 above)
Comment (atramos) [reply skepticoin/scripts/utils.py:110]: I'll move the `thread.daemon = True` to `balance.py`. That's where it's actually required.

**Comment by atramos** [comment]
Should be good now... please re-review

File: skepticoin/scripts/utils.py (hunk 1 above)
Comment (atramos) [reply skepticoin/scripts/utils.py]: Without this, the balance.py just hangs.

Comment (lk16) [reply skepticoin/scripts/utils.py]: What about using `thread.stop()` and `thread.join()`, similar to ~what we do~ what we [will be doing](https://github.com/skepticoin/skepticoin/pull/35/files#diff-67cc9641a1f9cd51b7bf7e4d5758ef7d155a517cfb13be11032dfd4b1480b6bcR147-R151) in MinerWatcher?

Comment (atramos) [reply skepticoin/scripts/utils.py]: That could work too

Comment (atramos) [reply skepticoin/scripts/utils.py]: changed it in 6d3a66274798b77d6d38c5b73bc397efea25c01b

**Comment by atramos** [comment]
@sashimi-houdini @lk16 this is ready to review

**Comment by lk16** [comment]
Please fix the flake8 error, then I'll merge it.

**Comment by atramos** [comment]
Please check now. I am getting 2 other mypy errors which are not related to my changes, not sure how to address those:
```
skepticoin/consensus.py:163: error: Returning Any from function declared to return "Block"
skepticoin/wallet.py:84: error: unused 'type: ignore' comment
```

**Pull request merged**